CovertInfiltration/Src/CovertInfiltration/Classes/UICovertActionsGeoscape.uc

```diff
@@ -874,15 +874,54 @@ simulated protected function UpdateProgressBar()
 		TotalDuration = class'X2StrategyGameRulesetDataStructures'.static.DifferenceInSeconds(CurrentAction.EndDateTime, CurrentAction.StartDateTime);
 		RemainingDuration = class'X2StrategyGameRulesetDataStructures'.static.DifferenceInSeconds(CurrentAction.EndDateTime, CurrentAction.GetCurrentTime());
 
+		ActionProgressBar.SetColor(class'UIUtilities_Colors'.const.COVERT_OPS_HTML_COLOR);
+		ActionProgressBar.SetBgColor(class'UIUtilities_Colors'.const.DISABLED_HTML_COLOR);
 		ActionProgressBar.Show();
 		ActionProgressBar.SetPercent(1 - RemainingDuration / TotalDuration);
 	}
+	else if (class'X2Helper_Infiltration'.static.IsInfiltrationAction(CurrentAction))
```

Comment:
Make a new function, `class'XComGameState_CovertActionExpirationManager'.static.IsExpirationAction(CurrentAction)`

Infiltration and expiration are not tied together, they are 2 different system, although we use them together